c_glib/arrow-glib/array-builder.h

```diff
@@ -22,70 +22,97 @@
 #include <arrow-glib/array.h>
 #include <arrow-glib/decimal.h>
 #include <arrow-glib/interval.h>
+#include <arrow-glib/visibility.h>
 
 G_BEGIN_DECLS
 
 #define GARROW_TYPE_ARRAY_BUILDER (garrow_array_builder_get_type())
+GARROW_EXPORT
 G_DECLARE_DERIVABLE_TYPE(
   GArrowArrayBuilder, garrow_array_builder, GARROW, ARRAY_BUILDER, GObject)
 struct _GArrowArrayBuilderClass
 {
   GObjectClass parent_class;
 };
 
+GARROW_EXPORT
 GArrowDataType *
 garrow_array_builder_get_value_data_type(GArrowArrayBuilder *builder);
+
+GARROW_EXPORT
 GArrowType
 garrow_array_builder_get_value_type(GArrowArrayBuilder *builder);
 
+GARROW_EXPORT
 GArrowArray *
 garrow_array_builder_finish(GArrowArrayBuilder *builder, GError **error);
 
 GARROW_AVAILABLE_IN_2_0
+GARROW_EXPORT
```

Comment:
Let's add this macro to `GARROW_AVAILABLE_IN_XXX`. GLib does it.

Reply:
Right. We need separated version macros per library.

Reply:
I thought I could have this PR only change the arrow-glib library and do the others in a follow up to keep the changes manageable, but the MinGW build fails with link errors like "redeclared without dllimport attribute after being referenced with dll linkage", so I might need to add the separated version macros as part of this PR too.

Reply:
I've changed the macro definitions to only add dllimport/dllexport for MSVC for now, so this works as currently only the arrow-glib library is built for MSVC. But I can follow up to also add them under MinGW GCC later after adding version macros per-library. Does that work for you or would you rather include the per-library macros in this PR?

Reply:
I like that idea, I've made #41681 for this and will make a PR for that before updating this PR.